pkg/operator/replacement_starter.go

```diff
@@ -0,0 +1,323 @@
+		operatorClient:               operatorClient,
+		authenticationOperatorClient: authenticationOperatorClient,
+		eventRecorder:                eventRecorder,
+		informers: []SimplifiedInformerFactory{
+			DynamicInformerFactoryAdapter(dynamicInformers), // we don't share the dynamic informers, but we only want to start when requested
+		},
+	}, nil
+}
+
+func CreateOperatorStarter(ctx context.Context, authOperatorInput *AuthenticationOperatorInput) (OperatorStarter, error) {
+	ret := &authenticationOperatorStarter{
+		eventRecorder: authOperatorInput.eventRecorder,
+		informers:     append([]SimplifiedInformerFactory{}, authOperatorInput.informers...),
+	}
+	kubeInformersForNamespaces := v1helpers.NewKubeInformersForNamespaces(
+		authOperatorInput.kubeClient,
+		"default",
+		"openshift-authentication",
+		"openshift-config",
+		"openshift-config-managed",
+		"openshift-oauth-apiserver",
+		"openshift-authentication-operator",
+		"", // an informer for non-namespaced resources
+		"kube-system",
+		libgoetcd.EtcdEndpointNamespace,
+	)
+	ret.informers = append(ret.informers, GeneratedNamespacedInformerFactoryAdapter(kubeInformersForNamespaces))
+	// resyncs in individual controller loops for this operator are driven by a duration based trigger independent of a resource resync.
+	// this allows us to resync essentially never, but reach out to external systems on a polling basis around one minute.
+	operatorConfigInformers := operatorinformer.NewSharedInformerFactory(authOperatorInput.operatorClient, 24*time.Hour)
+	ret.informers = append(ret.informers, GeneratedInformerFactoryAdapter(operatorConfigInformers))
+
+	versionRecorder := status.NewVersionGetter()
+	clusterOperator, err := authOperatorInput.configClient.ConfigV1().ClusterOperators().Get(ctx, "authentication", metav1.GetOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return nil, err
+	}
+	// perform version changes to the version getter prior to tying it up in the status controller
+	// via change-notification channel so that it only updates operator version in status once
+	// either of the workloads synces
+	for _, version := range clusterOperator.Status.Versions {
+		versionRecorder.SetVersion(version.Name, version.Version)
+	}
+	versionRecorder.SetVersion("operator", os.Getenv("OPERATOR_IMAGE_VERSION"))
+
+	resourceSyncer := resourcesynccontroller.NewResourceSyncController(
+		"oauth-server",
+		authOperatorInput.authenticationOperatorClient,
+		kubeInformersForNamespaces,
+		v1helpers.CachedSecretGetter(authOperatorInput.kubeClient.CoreV1(), kubeInformersForNamespaces),
+		v1helpers.CachedConfigMapGetter(authOperatorInput.kubeClient.CoreV1(), kubeInformersForNamespaces),
+		authOperatorInput.eventRecorder,
+	)
+	ret.controllerRunFns = append(ret.controllerRunFns, resourceSyncer.Run)
+	ret.controllerRunOnceFns = append(ret.controllerRunOnceFns, resourceSyncer.Sync)
+
+	return ret, nil
+}
+
+type GeneratedInformerFactory interface {
```

Comment:
why not to move `GeneratedInformerFactory` to a common place ? I think it will be needed for all operators, right?

Reply:
doing this in https://github.com/openshift/multi-operator-manager/pull/12 and will revendor